### DISCUSSION TITLE
Catch alarming arming issues

### DISCUSSION THREAD

**Discussion by jamesonuk** [tsightler/ring-mqtt#872] [Support Q&A]
Just wondering if there is anyway of catching issues when trying to set the alarm.
I have a HA automation which sets the alarm to arm when no-one is detected at home and recently this has started failing.   HA says it has done everything and it has all worked but nothing happens and there are no errors.

Turns out the battery had gone on a door sensor which wasn't set to bypass and this was preventing the alarming from setting but I had no visibility of this.   It was only when I was debugging it at home that I heard the base station saying sensors must be bypassed.

Is there anyway I could get feedback that the arm event has failed.   As far as HA is concerned it has fired off a message to MQTT to set the alarm so that it all good.   Obviously with MQTT there is no response but presumably this is caught within the integration and therefore could be published as a MQTT event which could then be used to trigger notifications that setting the alarm didn't work??

**Comment by tsightler** [comment]
Technically, it's not just "fire an MQTT message" it's also "received an MQTT state message" so, if an updated state message is not received, then that itself is the response.  I'm far from an expert in HA automations, but I think you can just add a delay and a conditional check to the action and that is all that is really needed.  For example, first action is arming, second action is delay for 5 seconds (or whatever time), 3rd action is condition that checks if alarm is in the requested armed state and sends a notification if it is not.

**Reply by jamesonuk** [reply]
That is what I have done but this obviously misses any context.   I haven't looked at how the Ring client handles it but I would have thought it returns a message that it couldn't arm as a sensor needs bypassing (possibly which sensor?).   I guess there could be other scenarios other than sensors needing bypass.    

If the error context is consumed then all I can tell from waiting is that it didn't work not why.   In this particular instance knowing that a sensor needed bypassing would have made me set that particular sensor to bypass always and to look at it when I got back home.

**Reply by tsightler** [reply]
Well, ring-mqtt does log the error message it receives if the arming fails, specifically if a sensor requires bypass you get this:

`Error: Failed to set alarm mode to "some".  Sensors may require bypass, which can only be done in the Ring app.`

**Reply by tsightler** [reply]
OK, technically this is the message from ring-client-api, not the Ring API directly.  All ring-client-api does is send the request to arm, with optional bypass sensors, then waits for the first returned message from the panel to see if the alarm enters the requested mode and returns that message if it doesn't.

To do anything more advanced, such as providing specific data, would require parsing the websocket message returned when arming for the specific context data.  When an arming requests fails due to sensor bypass, for example, it sends an impulse message that includes additional context data like this (this was in response to me attempting to arm with the front door open):

```
"impulse": {
    "v1": [
        {
            "data": {
                "devices": [
                    {
                        "name": "Front Entry Door",
                        "reasons": [
                            "fault"
                        ],
                        "zid": "1c822211-bb03-4f20-81a9-bba5ee11eba9"
                    }
                ],
                "mode": "some"
            },
            "impulseType": "security-panel.not-ready"
        }
    ]
}
```

Certainly it would be possible to send this raw data as part of the attribute topic (or some other error topic), but I'm not really sure how much value it really provides, I mean, you'd likely need to go into the Ring app anyway and, if the arming failed and you received a notification, that's what you would do anyway.   Not only that, but there could be plenty of other reasons arming might fail beyond bypass, maybe the hub is offline, maybe Ring API is down, etc.   Based on this, I don't really feel like it's worth the effort, as a failed arming will almost always need some manual investigation as to the cause.
